File: Backend/marketplace_server/entity/market_entry.py
from sqlalchemy import Column, BigInteger, String, DateTime, Boolean
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class MarketEntry(Base):
    __tablename__ = 'marketplace_content'
    nodeId = Column(BigInteger, primary_key=True)
    userId = Column(BigInteger, primary_key=True)
    name = Column(String)
    description = Column(String)
    created_at = Column(DateTime)
    updated_at = Column(DateTime)
    isArchived = Column(Boolean, default=False)

    # Add other fields as needed

File: serverless/rust/entity/market_entry.rs
use sea_orm::entity::prelude::*;
use sqlx::FromRow;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, FromRow)]
#[sea_orm(table_name = "marketplace_content")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub node_id: i64,
    #[sea_orm(primary_key)]
    pub user_id: i64,

    pub name: Option<String>,
    pub description: Option<String>,
    pub created_at: Option<DateTime>,
    pub updated_at: Option<DateTime>,
    pub is_archived: bool,
}

#[derive(Copy, Clone, Debug, EnumIter)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::users::Entity",
        from = "Column::UserId",
        to = "super::users::Column::Id"
    )]
    User,
}

impl Related<super::users::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::User.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
